version: '3.8'

services:
  frontend:
    build:
      context: ./frontend
      args:
        - VITE_API_URL=http://localhost:5000
    ports:
      - "3000:80"
    depends_on:
      - backend
    networks:
      - app-network
    environment:
      - VITE_API_URL=http://localhost:5000

  backend:
    build:
      context: ./backend
    ports:
      - "5000:5000"
    environment:
      - ARANGO_URL=http://arangodb:8529
      - ARANGO_DB_NAME=cag-pangaea-rs-v2
      - ARANGO_USER=root
      - ARANGO_PASS=root
      - NODE_PORT=5000
    depends_on:
      - arangodb
      - arangodb_restore
    restart: on-failure
    networks:
      - app-network

  arangodb:
    image: arangodb:latest
    ports:
      - "8529:8529"
    environment:
      - ARANGO_ROOT_PASSWORD=root
    volumes:
      - vesa_arango_data:/var/lib/arangodb3
      - ~/DB/arango-dump:/arango-dump
    networks:
      - app-network

  arangodb_restore:
    image: arangodb:latest
    depends_on:
      - arangodb
    volumes:
      - ~/DB/arango-dump:/arango-dump
      - ./import_dump.sh:/import_dump.sh
    environment:
      - ARANGO_ROOT_PASSWORD=root
      - ARANGO_DB_NAME=cag-pangaea-rs-v2
    command: /import_dump.sh
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  vesa_arango_data:
